File: 프로그래머스/0/181909. 접미사 배열/접미사 배열.java
import java.util.*;
class Solution {
    public String[] solution(String my_string) {
        int length = my_string.length();
        String[] answer = new String[length];  //접미사의 갯수가 문자열 길이와 같다
        
        for(int i=0; i<my_string.length(); i++){ //문자열 길이만큼 반복
            answer[i] = my_string.substring(i);  //answer배열에 자른 문자열 넣기
                                                 //"banana" 
                                                 //ex) i=0, a <- 인덱스번호로 자름
                                                 //    i=2, ana
        }
        Arrays.sort(answer); //배열의 원소를 오름차순으로 정렬
        return answer;
    }
}

File: 프로그래머스/0/181909. 접미사 배열/접미사 배열.py
def solution(my_string):
    answer = []                      # answer리스트 생성
    for i in range(len(my_string)):  # range():일정범위의 연속된 정수(my_string길이)를 반복
        answer.append(my_string[i:]) #리스트 answer에 인덱스 i자리부터 끝까지 문자열 추가
    answer.sort()                    #오름차순으로 정렬
    return answer